sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts

```diff
@@ -0,0 +1,798 @@
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    const pendingMSFAA = await db.msfaaNumber.save(
+      createFakeMSFAANumber(
+        { student },
+        {
+          msfaaState: MSFAAStates.Pending,
+          msfaaInitialValues: {
+            offeringIntensity: OfferingIntensity.partTime,
+          },
+        },
+      ),
+    );
+
+    const savedSFASIndividual = await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: { student: student },
+    });
+    const savedSFASPartTimeApplication = await db.sfasPartTimeApplications.save(
+      createFakeSFASPartTimeApplication(
+        { individual: savedSFASIndividual },
+        {
+          initialValues: {
+            startDate: getISODateOnlyString(firstLegacyApplicationStartDate),
+            endDate: getISODateOnlyString(firstLegacyApplicationEndDate),
+          },
+        },
+      ),
+    );
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+
+    // Fetch MSFAA Number for the disbursement schedule assigned to the student in SIMS.
+    const disbursementSchedule = await db.disbursementSchedule.findOne({
+      select: {
+        id: true,
+        msfaaNumber: { id: true, msfaaNumber: true },
+      },
+      relations: {
+        msfaaNumber: true,
+        studentAssessment: true,
+      },
+      where: {
+        studentAssessment: { id: application.currentAssessment.id },
+      },
+    });
+    // Fetch MSFAA Number for the student in SIMS.
+    const [createdMSFAANumber, cancelledMSFAANumber] =
+      await db.msfaaNumber.find({
+        select: {
+          id: true,
+          cancelledDate: true,
+          msfaaNumber: true,
+          referenceApplication: { id: true },
+          serviceProviderReceivedDate: true,
+          dateRequested: true,
+          dateSigned: true,
+        },
+        relations: {
+          referenceApplication: true,
+        },
+        where: {
+          student: { id: student.id },
+        },
+        order: {
+          id: "DESC",
+        },
+      });
+
+    // Assert MSFAA Number.
+    expect(disbursementSchedule.msfaaNumber.msfaaNumber).toBe(
+      savedSFASIndividual.partTimeMSFAANumber,
+    );
+    expect(createdMSFAANumber.msfaaNumber).toBe(
+      savedSFASIndividual.partTimeMSFAANumber,
+    );
+    expect(createdMSFAANumber).toEqual({
+      id: expect.any(Number),
+      msfaaNumber: disbursementSchedule.msfaaNumber.msfaaNumber,
+      dateSigned: savedSFASPartTimeApplication.endDate,
```

Comment:
We should validate if those values are the same as the SFAS, right?
Please use the `savedSFASPartTimeApplication` for asserts.